core/src/autogluon/core/utils/miscs.py

```diff
@@ -20,18 +20,14 @@ def in_ipynb():
         return False
 
 
-class warning_filter(object):
+class warning_filter(warnings.catch_warnings):
     def __enter__(self):
+        super().__enter__()
         warnings.filterwarnings("ignore", category=UserWarning)
         warnings.filterwarnings("ignore", category=FutureWarning)
         warnings.filterwarnings("ignore", category=DeprecationWarning)
         return self
 
-    def __exit__(self, *args):
```

Comment:
Should we still keep the `__exit__`?

Reply:
`__exit__` was altering global context. Therefore, if a different module was suppressing warnings, this code would globally reset the warnings so they are no longer suppressed. This led to MXNet spamming warnings about numpy deprecations after this class was used, because it overrode MXNet's custom warnings.

Now, since this inherits from `catch_warnings`, this reset is done automatically and does not reset global context, instead it resets to the global context, which is what we want.

Reply:
Thanks for the explanation. It looks good.